tests/bluechi_test/service.py

```diff
@@ -38,37 +46,61 @@ class Option(enum.Enum):
-        self.cp = configparser.ConfigParser()
-        # Enable case sensitivity
-        self.cp.optionxform = lambda option: option
-        self.name = name
+    """A base service class, which by default contains only predefined sections"""
+    def __init__(self, name: str = SERVICE_NAME, directory: str = SERVICE_DIRECTORY) -> None:
+        if not name:
+            raise Exception(f"Service name is '{name}', but cannot be empty!")
+
+        if not directory:
+            raise Exception(f"Service directory is '{directory}', but cannot be empty!")
+
+        self._svc_path = pathlib.Path(directory) / pathlib.Path(name)
+
+        self._cfg_parser = self._init_parser()
+
         for section in Section:
-            self.cp.add_section(section.value)
+            self._cfg_parser.add_section(section.value)
 
-    def __str__(self) -> str:
-        with io.StringIO() as content:
-            self.cp.write(content, space_around_delimiters=False)
-            return content.getvalue()
+    def _init_parser(self) -> configparser.ConfigParser:
+        cp = configparser.ConfigParser()
+        # Enable case sensitivity
+        cp.optionxform = lambda option: option
+        return cp
+
+    def get_path(self) -> pathlib.Path:
+        return self._svc_path
+
+    def get_name(self) -> str:
+        return self._svc_path.name
+
+    def get_directory(self) -> str:
+        return self._svc_path.parent
 
     def set_option(self, section: Section, name: Option, value: str) -> None:
-        self.cp.set(section.value, name.value, value)
+        self._cfg_parser.set(section.value, name.value, value)
 
     def get_option(self, section: Section, name: Option) -> Union[str, None]:
-        if self.cp.has_option(section.value, name.value):
-            return self.cp.get(section.value, name.value)
-
+        if self._cfg_parser.has_option(section.value, name.value):
+            return self._cfg_parser.get(section.value, name.value)
         return None
 
     def remove_option(self, section: Section, name: Option) -> None:
-        if self.cp.has_option(section.value, name.value):
-            self.cp.remove_option(section.value, name.value)
+        if self._cfg_parser.has_option(section.value, name.value):
+            self._cfg_parser.remove_option(section.value, name.value)
+
+    def to_string(self) -> str:
+        with io.StringIO() as content:
+            self._cfg_parser.write(content, space_around_delimiters=False)
+            return content.getvalue()
+
+    def from_string(self, content: str) -> None:
+        self._cfg_parser = self._init_parser()
+        self._cfg_parser.read_string(content)
 
 
 class SimpleService(Service):
     """A simple service, which is using /bin/true command"""
-    def __init__(self, name: str = "simple.service") -> None:
+    def __init__(self, name: str = "simple.service", directory: str = SERVICE_DIRECTORY) -> None:
         super(SimpleService, self).__init__(name)
```

Comment:
```suggestion
        super(SimpleService, self).__init__(name, directory)
```
The directory parameter needs to be passed in as well.

Reply:
Done